### PR TITLE
lang: add check that declared id == program id

### PR DESCRIPTION
**Pull request by paul-schaaf**

### CODE REVIEW

**Comment by paul-schaaf** [comment]
wasnt quite sure which category to put the error under. used `miscellaneous` for now

File: lang/src/error.rs [hunk 1]
```diff
@@ -16,6 +16,11 @@ pub const ERROR_CODE_OFFSET: u32 = 6000;
 /// by the [ERROR_CODE_OFFSET](crate::error::ERROR_CODE_OFFSET).
 #[error(offset = 0)]
 pub enum ErrorCode {
+    // Miscellaneous
+    /// 50 - The declared program id does not match actual program id
+    #[msg("The declared program id does not match the actual program id")]
+    DeclaredProgramIdMismatch = 50,
```
Comment (armaniferrante) [inline lang/src/error.rs:22]: Will this collide with solana_program error codes? Probably not?

File: tests/declare-id/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (armaniferrante) [inline tests/declare-id/package-lock.json:1]: file to be removed

File: tests/declare-id/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (armaniferrante) [inline tests/declare-id/yarn.lock:1]: file to be removed

File: tests/declare-id/programs/declare-id/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+use anchor_lang::prelude::*;
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS"); // intentionally different program id than the one defined in Anchor.toml
```
Comment (armaniferrante) [inline tests/declare-id/programs/declare-id/src/lib.rs:3]:
```suggestion
// Intentionally different program id than the one defined in Anchor.toml.
declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
```

File: lang/src/error.rs (hunk 1 above)
Comment (paul-schaaf) [reply lang/src/error.rs]: moved to `4100`

**Comment by paul-schaaf** [comment]
implemented review

**Pull request merged**